logback-core/src/main/java/ch/qos/logback/core/BasicStatusManager.java

```diff
@@ -57,10 +59,15 @@ public void add(Status newStatus) {
         // LBCORE-72: fire event before the count check
         fireStatusAddEvent(newStatus);
 
-        count++;
-        if (newStatus.getLevel() > level) {
-            level = newStatus.getLevel();
-        }
+        count.increment();
+        int newLevel = newStatus.getLevel();
+        int currentLevel;
+        do {
+            currentLevel = level.get();
+            if (newLevel <= currentLevel) {
+                break;
+            }
+        } while (!level.compareAndSet(currentLevel, newLevel));
 
         synchronized (statusListLock) {
```

Comment:
Here's an alternate approach to CAS

```suggestion
        synchronized (statusListLock) {
            count.increment();
            int newLevel = newStatus.getLevel();
            if (newLevel > level.get()) {
                level.set(newLevel);
            }

```